### DISCUSSION TITLE
olivetin/webUiSettings.json

### DISCUSSION THREAD

**Discussion by andresanz** [OliveTin/OliveTin#80] [Q&A] [answered]
So try as I might, olivetin is not starting properly - it creates an olivetin folder in /var/www, but I moved it to a url within /var/www/html.  webUiSettings.json does not exist anywhere - help!
![image](https://user-images.githubusercontent.com/8781111/197071902-d15758cf-42a6-4894-a001-3a3e4f2e3e6f.png)

**Comment by jamesread** [comment]
Hey @andresanz , sorry that you've got this issue, I'm sure I can help with that. 

What's most likely happening here is that the OliveTin webui is loading, but it's can't make the rest API call to the OliveTin API (which should default to "/api", and the absolute address would be http://myserver:1337/api), but your webserver is giving it another page (probably a 404). 

I can see that you're hosting the webui at "/utilities/olivetin/", which I call "path based" reverse proxying. In order to do that, you will need to tell OliveTin exactly what "base URL" to use - so that "utilities/olivetin/api" works - The config setting is called `externalRestAddress`. If you're using nginx or caddy, there are some instructions here that should help; https://docs.olivetin.app/reverse-proxies.html

If you're not, could you tell me which reverse proxy it is that you're using, and post your config? It would make for good documentation in the future. Also, which version are you using? Weirdly the footer shows "Version: ?". Hope this helps!

**Comment by andresanz** [comment, marked as answer]
I put my response into the discord server - hope its more helpful there

André Sanz | ***@***.*** | 914-659-6430

**Reply by andresanz** [reply]
https://discord.com/channels/846737624960860180/869602195216142376/1033141103395880980

**Comment by jamesread** [comment]
@andresanz , am I right in thinking that we solved your issue via Discord? If so, could you mark one of the comments as the answer to close this thread? Thanks. :-)
